Assets/Dev/Si Wai/PermanentPlayerStatistic.cs

```diff
@@ -0,0 +1,106 @@
+    public float Health {
+        get => health;
+        set => health = value > 0 ? value : 0;
+    }
+    private int crumbs;
+    public int Crumbs
+    {
+        get => crumbs;
+        set => crumbs = value > 0 ? value : 0;
+    }
+    public PermanentStatistic AttackSpeedMultiplier = new();
+    public PermanentStatistic AttackDamageMultiplier = new();
+    public PermanentStatistic DodgeCooldown = new();
+    public PermanentStatistic DoubleJumpsCount = new();
+    protected override string Prefix => "permanentPlayerStatistic";
+
+    public override void Init() {
+        // using json strings to save the stats
+        string json;
+
+        json = JsonUtility.ToJson(Speed.ListWithModifications());
+        Add("speed", json);
+
+        json = JsonUtility.ToJson(JumpForce.ListWithModifications());
+        Add("jumpForce", json);
+
+        json = JsonUtility.ToJson(MaxHealth.ListWithModifications());
+        Add("maxHealth", json);
+
+        Add("health", Health);
+        Add("crumbs", Crumbs);
+
+        json = JsonUtility.ToJson(AttackSpeedMultiplier.ListWithModifications());
+        Add("attackSpeedMultiplier", json);
+
+        json = JsonUtility.ToJson(AttackDamageMultiplier.ListWithModifications());
+        Add("attackDamageMultiplier", json);
+
+        json = JsonUtility.ToJson(DodgeCooldown.ListWithModifications());
+        Add("dodgeCooldown", json);
+
+        json = JsonUtility.ToJson(DoubleJumpsCount.ListWithModifications());
+        Add("doubleJumpsCount", json);
+    }
+
+    public void SaveSpeed() {
+        string json = JsonUtility.ToJson(Speed.ListWithModifications());
+        GlobalReference.PermanentPlayerStatistic.Set("speed", json);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveJumpForce() {
+        string json = JsonUtility.ToJson(JumpForce.ListWithModifications());
+        GlobalReference.PermanentPlayerStatistic.Set("jumpForce", json);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveMaxHealth() {
+        string json = JsonUtility.ToJson(MaxHealth.ListWithModifications());
+        GlobalReference.PermanentPlayerStatistic.Set("maxHealth", json);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveHealth() {
+        GlobalReference.PermanentPlayerStatistic.Set("health", Health);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveCrumbs() {
+        GlobalReference.PermanentPlayerStatistic.Set("crumbs", Crumbs);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveAttackSpeedMulitplier() {
+        string json = JsonUtility.ToJson(AttackSpeedMultiplier.ListWithModifications());
+        GlobalReference.PermanentPlayerStatistic.Set("attackSpeedMultiplier", json);
+        GlobalReference.PermanentPlayerStatistic.SaveAll();
+    }
+
+    public void SaveAttackDamageMultiplier() {
```

Comment:
Also, why are you rewriting the exact same method over and over again.
If we have a stat, whe also have to rewrite this method.
If you write code, you always need to think about scalability. And this is simply not, since we have to create a new method each single time.

Instead what you can do is give a name to the permanent stat, like, that string value. And then just have 1 method in the permanent stat that does this.

So instead of `public PermanentStatistic DodgeCooldown = new();` You get
`public PermanentStatistic DodgeCooldown = new("dodgeCooldown");`
And instead of creating a whole new method called `SaveDodgeCooldown()`
You could just do `DodgeCooldown.Save()`

Reply:
(and yes again, Crumbs is here the exception)